src/SFML/Graphics/Shape.cpp

```diff
@@ -275,14 +255,21 @@ void Shape::updateOutline()
         Vector2f p2 = m_vertices[index + 1].position;
 
         // Compute their normal
-        Vector2f n1 = computeNormal(p0, p1);
-        Vector2f n2 = computeNormal(p1, p2);
+        const auto findNormal = [](const sf::Vector2f& vector)
+        {
+            if (vector.lengthSq() == 0.f)
+                return vector;
+            return vector.perpendicular().normalized();
```

Comment:
I believe that this PR pessimizes the performance of this code, as now we're computing the length twice. Maybe it is worthwhile only using the `.dot` member function and keeping the older code for the perpendicular+normalized?

Reply:
That's fair. I just pushed a simpler version of this PR.